### PR TITLE
Fix printing of tab-indented files

### LINKED ISSUE #1964: Whitespace indentation breaking
| Subject        | Details                                                      |
| :------------- | :----------------------------------------------------------- |
| PHP version    | e.g. PHP 7.3 (invoke `php -v`)                               |
| Full Command   | e.g. vendor/bin/rector process /src --set=cakephp34  | 

Using latest dev-master branch.

## Current Behavior 

Dry-Run shows what will happen
```
                /** @var \Cake\Console\ConsoleInputSubcommand[] $subcommands */
-               $subcommands = $parser->subcommands();
+  $subcommands = $parser->subcommands();
```
and
```
        /**
-        * Get the option parser
-        *
-        * @return \Cake\Console\ConsoleOptionParser
-        */
-       public function getOptionParser() {
+  * Get the option parser
+  *
+  * @return \Cake\Console\ConsoleOptionParser
+  */
+ public function getOptionParser() {
```

Once applied it actually destroys the indentation (2 tabs => 2 spaces).

It looks like the docblock before could cause this.

## Expected Behavior

Current indentation/whitespace should not be touched/modified.
Especially, since in this case there is not even a change here done.

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #1964

### CODE REVIEW

File: abz/SomeFile.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App;
+
+class Test {
```
Comment (dereuromark) [inline abz/SomeFile.php]: We could call it TabIndentationTest to be more speaking.

**Comment by dereuromark** [comment]
I would also keep an line comment inside a function to be sure that one is also covered.

**Comment by TomasVotruba** [comment]
One step at a time

File: src/PhpParser/Printer/BetterStandardPrinter.php [hunk 2]
```diff
@@ -38,6 +46,14 @@ public function __construct(array $options = [])
         $this->insertionMap['Expr_Closure->returnType'] = [')', false, ': ', null];
     }
 
+    public function printFormatPreserving(array $stmts, array $origStmts, array $origTokens): string
+    {
+        // detect per print
+        $this->detectTabOrSpaceIndentCharacter($stmts);
```
Comment (TomasVotruba) [inline src/PhpParser/Printer/BetterStandardPrinter.php]: @dereuromark Here is the important change. PhpParser prints everything with spaces-only by default. So there was actually 1 indent in the code, just not 1 tab, but 1 space.

Detecting spaces per line would be slow and not very useful, so 1st newlinew indent kind is taken into account - see `detectTabOrSpaceIndentCharacter()` bellow.

What do you think?

Comment (dereuromark) [reply src/PhpParser/Printer/BetterStandardPrinter.php]: Sounds reasonable to me :)

Comment (TomasVotruba) [reply src/PhpParser/Printer/BetterStandardPrinter.php]: Ok :), let's go with then. If anything breaks in the future, we know wher to look for.

**Comment by dereuromark** [comment]
Wow, awesome :tada: Thanks a lot!

**Pull request merged**